extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/execution/Selector.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.k8s.overlord.execution;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.indexing.common.task.Task;
+import org.apache.druid.query.DruidMetrics;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents a condition-based selector that evaluates whether a given task meets specified criteria.
+ * The selector uses conditions defined on context tags and task fields to determine if a task matches.
+ */
+public class Selector
+{
+  private final String name;
+  private final Map<String, Set<String>> cxtTagsConditions;
+  private final Map<String, Set<String>> taskFieldsConditions;
+
+  /**
+   * Creates a selector with specified conditions for context tags and task fields.
+   *
+   * @param name                 the identifier representing the outcome when a task matches the conditions
+   * @param cxtTagsConditions    conditions on context tags
+   * @param taskFieldsConditions conditions on task fields
+   */
+  @JsonCreator
+  public Selector(
+      @JsonProperty("name") String name,
```

Comment:
Is `name` right here? It makes it feel like there can only be one of these for each category. Can we make it `categoryName` instead?

Reply:
As this `Selector` class will be used for not only category but may be task laning work in the future, so I make it not tied to `categoryName` thus it can be reused easily for other purpose.

Reply:
Ok, I think `name` is still a fairly confusing term because it seems like the name for the selector and not the category to select. Would a name like `selectionKey` or `selectKey` work better? That way it's general purpose enough.

Reply:
updated with selectionKey